sale_stock_line_customer_ref/models/stock_package_level.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class StockPackageLevel(models.Model):
+    _inherit = "stock.package_level"
+
+    has_customer_ref = fields.Boolean(
+        compute="_compute_has_customer_ref",
+        help="Technical field to display 'Customer Ref' column in a package level.",
+    )
+
+    @api.depends("move_ids.customer_ref")
+    def _compute_has_customer_ref(self):
+        for pl in self:
+            pl.has_customer_ref = any(pl.move_ids.mapped("customer_ref")) or any(
```

Comment:
This is going to loop on all records every time.
In such cases I normally prefer a look + break on the 1st one matching.
Here's a shortcut:
```
next((x for x in pl.move_ids if x.customer_ref), False)
```
See https://stackoverflow.com/questions/2361426/get-the-first-item-from-an-iterable-that-matches-a-condition

Reply:
Indeed, knowing that we are just saving CPU time here, we still fetch the same amount of data (IO) even if we break on the first occurrence, but it's better than nothing.

Reply:
Small test added regarding `picking.has_customer_ref`, a bit lazy to add one for the package level... :)